### ISSUE TITLE
order state before submitting to finalize

### ISSUE THREAD

**Issue by dolphinsd** [PKISharp/ACMESharpCore#26]
Per the docs, order should be in Ready state before finalization.
See https://tools.ietf.org/html/draft-ietf-acme-acme-12#section-7.1.3

The code in AcmeHostedServices checks for PendingState, not readystate.

if (AcmeState.PendingStatus == _state.Order.Payload.Status)
            {
                _logger.LogInformation("Generating CSR");

**Comment by WouterTinus** [comment]
I'm not familiar with the section of code that you are referring to, but "Generating CSR" suggests that that the order cannot possibly be "ready" yet, because the CSR has to be submitted before the certificate can be generated.

**Comment by dolphinsd** [comment]
@WouterTinus , the code is from AcmeHostedServices sample, search for "Generating CSR" 
We traced the steps, and in our case, order is in ReadyState before generating csr. Check the states in the docs for explanation.

**Comment by WouterTinus** [comment]
You are right, this looks like a bug. Probably it works fine as long as the locally cached order object is not updated after submitting the authorization response(s), *except* when all host names just happened to have valid authorizations already (i.e. from a previous run) and so the new order immediately gets to the `ready` state.

**Comment by omnibasissupport** [comment]
Maybe related
https://github.com/PKISharp/ACMESharpCore/issues/34
